File: docs/breez-sdk/snippets/go/webhooks.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-spark-go/breez_sdk_spark"
)

func RegisterWebhook(sdk *breez_sdk_spark.BreezSdk) (*breez_sdk_spark.RegisterWebhookResponse, error) {
	// ANCHOR: register-webhook
	response, err := sdk.RegisterWebhook(breez_sdk_spark.RegisterWebhookRequest{
		Url:    "https://example.com/webhook",
		Secret: "your-webhook-secret",
		EventTypes: []breez_sdk_spark.WebhookEventType{
			breez_sdk_spark.WebhookEventTypeLightningReceiveFinished{},
			breez_sdk_spark.WebhookEventTypeLightningSendFinished{},
		},
	})
	if err != nil {
		return nil, err
	}

	log.Printf("Webhook registered with ID: %v", response.WebhookId)
	// ANCHOR_END: register-webhook
	return &response, nil
}

func UnregisterWebhook(sdk *breez_sdk_spark.BreezSdk) error {
	// ANCHOR: unregister-webhook
	webhookId := "webhook-id"
	err := sdk.UnregisterWebhook(breez_sdk_spark.UnregisterWebhookRequest{
		WebhookId: webhookId,
	})
	if err != nil {
		return err
	}

	log.Printf("Webhook unregistered")
	// ANCHOR_END: unregister-webhook
	return nil
}

func ListWebhooks(sdk *breez_sdk_spark.BreezSdk) ([]breez_sdk_spark.Webhook, error) {
	// ANCHOR: list-webhooks
	webhooks, err := sdk.ListWebhooks()
	if err != nil {
		return nil, err
	}

	for _, webhook := range webhooks {
		log.Printf("Webhook: id=%v, url=%v, events=%v", webhook.Id, webhook.Url, webhook.EventTypes)
	}
	// ANCHOR_END: list-webhooks
	return webhooks, nil
}

File: docs/breez-sdk/snippets/rust/src/webhooks.rs
use anyhow::Result;
use breez_sdk_spark::*;
use log::info;

pub(crate) async fn register_webhook(sdk: &BreezSdk) -> Result<()> {
    // ANCHOR: register-webhook
    let response = sdk
        .register_webhook(RegisterWebhookRequest {
            url: "https://example.com/webhook".to_string(),
            secret: "your-webhook-secret".to_string(),
            event_types: vec![
                WebhookEventType::LightningReceiveFinished,
                WebhookEventType::LightningSendFinished,
            ],
        })
        .await?;
    info!("Webhook registered with ID: {}", response.webhook_id);
    // ANCHOR_END: register-webhook
    Ok(())
}

pub(crate) async fn unregister_webhook(sdk: &BreezSdk) -> Result<()> {
    // ANCHOR: unregister-webhook
    let webhook_id = "webhook-id".to_string();
    sdk.unregister_webhook(UnregisterWebhookRequest { webhook_id })
        .await?;
    info!("Webhook unregistered");
    // ANCHOR_END: unregister-webhook
    Ok(())
}

pub(crate) async fn list_webhooks(sdk: &BreezSdk) -> Result<()> {
    // ANCHOR: list-webhooks
    let webhooks = sdk.list_webhooks().await?;
    for webhook in webhooks {
        info!(
            "Webhook: id={}, url={}, events={:?}",
            webhook.id, webhook.url, webhook.event_types
        );
    }
    // ANCHOR_END: list-webhooks
    Ok(())
}
